tapdb/supply_commit.go

```diff
@@ -0,0 +1,1303 @@
+			OutputKey:      outputKey.SerializeCompressed(), //nolint:lll
+			SupplyRootHash: nil,
+			SupplyRootSum:  sql.NullInt64{},
+			OutputIndex:    sqlInt32(outputIndex),
+		})
+		if err != nil {
+			return fmt.Errorf("failed to insert new supply "+
+				"commitment: %w", err)
+		}
+
+		// Update the transition record to link to the new commitment ID
+		// and the pending chain transaction ID in a single query.
+		err = db.UpdateSupplyCommitTransitionCommitment(
+			ctx, UpdateSupplyCommitTransitionCommitmentParams{
+				NewCommitmentID:    sqlInt64(newCommitmentID),
+				PendingCommitTxnID: sqlInt64(chainTxID),
+				TransitionID:       pendingTransition.TransitionID, //nolint:lll
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to update transition "+
+				"commitment: %w", err)
+		}
+
+		// As the final step, we'll now update the state on disk to move
+		// broadcast the commit txn we just signed.
+		// to the broadcast state. This ensures that on restart we'll
+		broadcastStateName, err := stateToDBString(
+			&supplycommit.CommitBroadcastState{},
+		)
+		if err != nil {
+			return fmt.Errorf("error getting broadcast state "+
+				"name: %w", err)
+		}
+		// We only update the state name here, leaving the commitment ID
+		// as is (by passing NULL).
+		_, err = db.UpsertSupplyCommitStateMachine(
+			ctx, SupplyCommitMachineParams{
+				GroupKey:  groupKeyBytes,
+				StateName: sqlStr(broadcastStateName),
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to update state machine "+
+				"state: %w", err)
+		}
+
+		return nil
+	})
+}
+
+// CommitState commits the state of the state machine to disk.
+func (s *SupplyCommitMachine) CommitState(ctx context.Context,
+	assetSpec asset.Specifier, state supplycommit.State) error {
+
+	groupKey := assetSpec.UnwrapGroupKeyToPtr()
+	if groupKey == nil {
+		return ErrMissingGroupKey
+	}
+	groupKeyBytes := groupKey.SerializeCompressed()
+
+	newStateName, err := stateToDBString(state)
+	if err != nil {
+		return fmt.Errorf("failed to map state to string: %w", err)
+	}
+
+	var writeTx SupplyCommitTxOptions
+	return s.db.ExecTx(ctx, &writeTx, func(db SupplyCommitStore) error {
+		// We only update the state name here, leaving the commitment ID
+		// as is (by passing NULL).
+		_, err = db.UpsertSupplyCommitStateMachine(
+			ctx, SupplyCommitMachineParams{
+				GroupKey:  groupKeyBytes,
+				StateName: sqlStr(newStateName),
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to update state machine "+
+				"state: %w", err)
+		}
+		return nil
+	})
+}
+
+// fetchCommitment is a helper to fetch and reconstruct a RootCommitment and
+// its associated chain confirmation details.
+func fetchCommitment(ctx context.Context, db SupplyCommitStore,
+	commitID sql.NullInt64, groupKeyBytes []byte,
+) (lfn.Option[supplycommit.RootCommitment], lfn.Option[commitmentChainInfo], error) { //nolint:lll
+
+	if !commitID.Valid {
+		return lfn.None[supplycommit.RootCommitment](),
```

Comment:
We could create those `lfn.None[supplycommit.RootCommitment]()` and `lfn.None[commitmentChainInfo]()` as variables above, since they're used so many times across this function.